src/settingsholder.cpp

```diff
@@ -31,21 +37,37 @@ SettingsHolder* SettingsHolder::instance() {
   return s_instance;
 }
 
-#ifndef UNIT_TEST
-const QSettings::Format MozFormat = QSettings::registerFormat(
-    "moz", CryptoSettings::readFile, CryptoSettings::writeFile);
-#endif
-
 SettingsHolder::SettingsHolder()
-    :
+    : m_settings(MozFormat, QSettings::UserScope,
 #ifndef UNIT_TEST
-      m_settings(MozFormat, QSettings::UserScope, "mozilla", "vpn")
+                 "mozilla",
 #else
-      m_settings("mozilla_testing", "vpn")
+                 "mozilla_testing",
 #endif
-{
+                 "vpn") {
   MVPN_COUNT_CTOR(SettingsHolder);
 
+  if (QFile::exists(journalSettingFileName())) {
+    QString journalSettingFile(journalSettingFileName());
+    logger.warning() << "journal file exists" << journalSettingFile;
```

Comment:
If a file exists it means the app was closed unexpectedly with a transaction was happening. Correct?

Does this really need to be a warning? Also, can you add to the log message that a recovery process will start?

Reply:
Yes, let's replace `warning` with `info`